### PR TITLE
fix: show request membership only for forbidden access

### LINKED ISSUE #1506: Do not show request membership for ITEM_NOT_FOUND
The button should be displayed only for MEMBER_CANNOT_ACCESS

### PR DESCRIPTION
**Pull request by pyphilia**
fix #1506 

question: maybe we want to show `item not found 404` for the other cases? instead of showing cannot have access?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,11 +20,11 @@
     "@emotion/styled": "11.13.0",
     "@graasp/chatbox": "3.3.0",
     "@graasp/map": "1.19.0",
-    "@graasp/query-client": "3.26.0",
+    "@graasp/query-client": "4.0.0",
     "@graasp/sdk": "4.31.0",
     "@graasp/stylis-plugin-rtl": "2.2.0",
     "@graasp/translations": "1.39.0",
-    "@graasp/ui": "5.2.0",
+    "@graasp/ui": "github:graasp/graasp-ui#forbidden-request-access",
```
Comment (ReidyT) [inline package.json:27]: reminder ^^

File: src/components/pages/item/accessWrapper/ItemAccessWrapper.tsx [hunk 2]
```diff
@@ -45,6 +49,9 @@ const ItemAccessWrapper = (): JSX.Element => {
   return (
     <ItemLoginWrapper
       item={item}
+      itemErrorStatusCode={
+        (axios.isAxiosError(itemError) && itemError.status) || null
```
Comment (ReidyT) [inline src/components/pages/item/accessWrapper/ItemAccessWrapper.tsx:53]: Should we extract this?

```
const errorStatusCode = (axios.isAxiosError(itemError) && itemError.status) || null
```

**Review by ReidyT — APPROVED** [review]
LGTM 🌝

File: cypress/e2e/item/view/notFound.cy.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+describe('Item does not exist', () => {
+  it('Show forbidden message and cannot request membership', () => {
+    cy.setUpApi();
+    cy.visit(buildItemPath(v4()));
+
+    // no add button
+    cy.get(`#${ITEM_LOGIN_SCREEN_FORBIDDEN_ID}`).should('be.visible');
+
+    // menu item only contains flag
+    cy.get(`#${REQUEST_MEMBERSHIP_BUTTON_ID}`).should('not.exist');
```
Comment (spaenleh) [inline cypress/e2e/item/view/notFound.cy.ts:18]: I don't think the comments make sens here ...

**Review by spaenleh — APPROVED** [review]
Looks good, just adapt the comments.

**Pull request merged**